Add skypilot agent skill (#8823)
skypilot-org/skypilot: skills/skypilot/SKILL.md

---
name: skypilot

> description: Cloud GPU orchestration for AI workloads. Use when launching GPU clusters, running training jobs, serving models, or managing compute across 25+ clouds and Kubernetes.

Reviewer: ```suggestion
description: Orchestrate for AI workloads on any infra, including clouds, Kubernetes clusters, Slurm, or on-prem clusters. Use when launching resources, running interactive development/sandbox, training, or serving models. It can manage compute across 25+ clouds, Kubernetes clusters, Slurm clusters and on-prem clusters.
```

version: 1.0.0
author: SkyPilot Authors
license: Apache-2.0
tags: [Infrastructure, Multi-Cloud, GPU, Kubernetes, Training, Serving, Cost Optimization, SkyPilot]
dependencies: [skypilot>=0.10.0]
---

# SkyPilot Skill

SkyPilot is a unified framework to run AI workloads on any cloud or Kubernetes. It provides a single interface to launch clusters, run training jobs, and serve models across 25+ clouds (AWS, GCP, Azure, Lambda, RunPod, Kubernetes, and more).

## When to Use SkyPilot

**Use SkyPilot when you need to:**

> - Manage compute resources on any cloud or Kubernetes cluster

Reviewer: ```suggestion
- Manage compute resources on any cloud, Slurm, or Kubernetes cluster
```

> - Launch GPU/TPU clusters (H100, A100, V100, etc.) on any cloud

Reviewer: ```suggestion
- Launch CPU/GPU/TPU (GB300, GB200, B200, H200, H100, etc.) on any cloud, Kubernetes or Slurm.
```

```
Ask the user which clouds they need if unclear.

**Step 2: Check cloud credentials**
```bash
sky check
```
This shows which clouds are configured. If the user's target cloud is not enabled, guide them through credential setup (see [Troubleshooting](references/troubleshooting.md#1-installation-and-credentials)).

## Quick Start

```bash
# List available GPUs
sky gpus list

# Launch a GPU cluster
sky launch -c mycluster --gpus H100 -- nvidia-smi

# Run a task from YAML
sky launch -c mycluster task.yaml

# SSH into cluster
ssh mycluster

# Tear down
sky down mycluster
```

## Task YAML Structure

The task YAML is SkyPilot's primary interface. All fields are optional.

```yaml
# task.yaml
name: my-training-job

# Local directory to sync to remote ~/sky_workdir
workdir: .

# Number of nodes (for distributed training)
num_nodes: 1

resources:
  # GPU/TPU accelerators (SkyPilot auto-selects the cheapest cloud/region)
  accelerators: H100:8
  # Optional: pin to a specific cloud/region if needed
  # infra: aws  # or aws/us-east-1, k8s, ssh/my-pool
  # Use spot instances for cost savings
  use_spot: false
  # Disk size in GB
  disk_size: 256
  # Open ports for serving

>   ports: 8080

Reviewer: I don't think these are critical. We can leave them out for now.

> 

Reviewer: add infra field and say that it will automatically failover if left out

  BATCH_SIZE: 32

# Secrets (redacted in logs/dashboard)
secrets:
  HF_TOKEN: my-huggingface-token

# Setup: runs once on cluster creation, cached on reuse
setup: |
  pip install torch transformers

# Run: the main command
run: |
  python train.py --model $MODEL_NAME --batch-size $BATCH_SIZE
```

## Key CLI Commands

| Command | Description |
|---------|-------------|
| `sky launch` | Launch a cluster or run a task |
| `sky exec` | Execute a task on existing cluster |
| `sky status` | Show cluster status |
| `sky stop` / `sky start` | Stop/start clusters (preserves disk) |
| `sky down` | Tear down clusters |
| `sky autostop` | Configure auto-stop after idle time |
| `sky queue` | Show job queue on a cluster |
| `sky logs` | Stream job logs |
| `sky cancel` | Cancel running jobs |
| `sky gpus list` | Show GPU availability and pricing |
| `sky check` | Check cloud credentials |
| `sky cost-report` | Show cluster cost report |
| `sky jobs launch` | Launch a managed job (auto-recovery) |
| `sky jobs queue` | Show managed job queue |
| `sky serve up` | Start a model serving service |
| `sky serve status` | Show service status |
| `sky api start` | Start the SkyPilot API server |

> | `sky dashboard` | Open the SkyPilot web dashboard |

Reviewer: Can we categorize the commands into `SkyPilot clusters`, `managed jobs`, and `SkyServe`

    - infra: gcp/us-central1
      accelerators: H100:8
    - infra: aws/us-west-2
      accelerators: A100-80GB:8
```

Only set `infra:` when the user explicitly says something like "use AWS" or "run on GCP us-central1":

```yaml
resources:
  infra: aws             # User asked for AWS specifically
  accelerators: H100:8
```

## Cluster Lifecycle

```bash
# Launch and run a task
sky launch -c mycluster task.yaml

# Re-run a different task on the same cluster (fast, skips provisioning)
sky exec mycluster another_task.yaml

# Run an inline command
sky exec mycluster -- python train.py --epochs 10

# Set autostop (stop after 30 min idle, preserving disk)
sky autostop mycluster -i 30

# Set autodown (tear down after 30 min idle)
sky autostop mycluster -i 30 --down

# Stop to save costs, restart later
sky stop mycluster
sky start mycluster

# Tear down completely
sky down mycluster

# Tear down all clusters
sky down -a
```

## Managed Jobs (Auto-Recovery from Spot Preemptions)

Use `sky jobs launch` for long-running jobs that should survive spot preemptions:

```yaml
# managed-job.yaml
name: training-with-recovery

resources:
  accelerators: A100:8
  use_spot: true
  job_recovery:

>     strategy: FAILOVER             # or EAGER_NEXT_REGION

Reviewer: I don't think we need to expose this for now
